knowledge_repo/postprocessors/extract_images.py

```diff
@@ -50,14 +50,17 @@ def update_thumbnail_uri(cls, kp, images, image_mapping):
     def find_images(cls, md):
         images = []
         images.extend(cls.collect_images_for_pattern(
-            md, '<img.*src=[\'"](.*?)[\'"].*>'))
-        images.extend(cls.collect_images_for_pattern(md, '\!\[.*\]\((.*)\)'))
+            md,
+            r'<img\s+(?:\w+(?:=([\'\"])?(?(1)(?:(?!\1).)*?\1|[^>]*?))?\s+?)*src=([\'\"])(?P<src>(?:(?!\2).)*?)\2(?:\s+\w+(?:=([\'\"])?(?(1)(?:(?!\4).)*?\4|[^>]*?))?)*\s*\/?>'
+        ))
```

Comment:
ooooh could we have a comment on what this regexp is supposed to match for future reference? (I'm very bad at reading regexp)